File: backend/jeerai-backend/src/main/java/com/jeerai/backend/model/Project.java
package com.jeerai.backend.model;
import java.time.Instant;
import java.util.ArrayList;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Project {
    private String id;
    private String key;
    private String name;
    private String description;
    private User lead;
    private List<User> members = new ArrayList<>();
    private Instant createdAt;
    private Instant updatedAt;
    private String workspaceId;
}

File: frontend/src/types/project.ts
import type { User } from './user';
import type { WorkspaceRole } from './workspace';
export interface Project {
  id: string;
  key: string;
  name: string;
  description: string;
  lead: User;
  members: User[];
  createdAt: string;
  updatedAt: string;
  workspaceId?: string;
}
export interface Sprint {
  id: string;
  name: string;
  projectId: string;
  startDate: string;
  endDate: string;
  isActive: boolean;
}
export type ProjectPermissionKey = 'CREATE_ISSUES' | 'EDIT_ISSUES' | 'DELETE_ISSUES' | 'MANAGE_PROJECT' | 'VIEW_ANALYTICS';
export interface ProjectPermissions {
  projectId: string;
  permissions: Record<WorkspaceRole, Record<ProjectPermissionKey, boolean>>;
}
